src/v/cloud_storage/remote.cc

```diff
@@ -101,9 +101,9 @@ static error_outcome categorize_error(
         // - any filesystem error
         // - broken-pipe
         // - any other network error (no memory, bad socket, etc)
-        if (auto code = cerr.code(); code.value() != ECONNREFUSED
-                                     && code.value() != ENETUNREACH
-                                     && code.value() != ECONNRESET) {
+        if (auto code = cerr.code();
+            code.value() != ECONNREFUSED && code.value() != ENETUNREACH
+            && code.value() != ETIMEDOUT && code.value() != ECONNRESET) {
```

Comment:
nice!

Reply:
should we update the comment above?

Reply:
question, can we do the opposite and instead retry-by-default? 

cc: @Lazin ?

Reply:
I think that this error will be handled on line 116
the code in `src/v/rpc/transport` invokes connection using ss::with_timeout which throws this exception

Reply:
@senior7515 I think the existing comment is fine, we already mention that we should be handling the timeout as retryable, but the errno wasn't  handling a regular timeout, just ENETUNREACH which is a No Route to Host error. An ETIMEDOUT could happen if the socket doesn't get any response from the remote host under the configured socket timeout which seems to be 5 seconds for the http::client .

Reply:
I think I should have added to the cover letter some of the information I dumped in the issue: https://github.com/vectorizedio/redpanda/issues/2094

Reply:
Oh, I think I understand now. If the error can be propagated as system_error we should whitelist it here.